platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/collections/LongListDisk.java

```diff
@@ -40,65 +91,128 @@ public class LongListDisk extends LongList {
      * @throws IOException If there was a problem reading the file
      */
     public LongListDisk(final Path file) throws IOException {
-        super(FileChannel.open(file, StandardOpenOption.CREATE, StandardOpenOption.READ, StandardOpenOption.WRITE));
-        this.file = file;
+        this(file, DEFAULT_RESERVED_BUFFER_LENGTH);
+    }
+
+    LongListDisk(final Path file, final long reservedBufferLength) throws IOException {
+        super(file, reservedBufferLength);
+        freeChunks = new ConcurrentLinkedDeque<>();
+    }
+
+    @Override
+    protected void readBodyFromFileChannelOnInit(final String sourceFileName, final FileChannel fileChannel)
+            throws IOException {
+        Path tempFile = createTempFile(sourceFileName);
+        // create temporary file for writing
+        // the warning is suppressed because the file is not supposed to be closed
+        // as this implementation uses a file channel from it.
+        try (final RandomAccessFile rf = new RandomAccessFile(tempFile.toFile(), "rw")) {
+            // ensure that the amount of disk space is enough
+            // two additional chunks are required to accommodate "compressed" first and last chunks in the original file
+            rf.setLength(fileChannel.size() + 2L * memoryChunkSize);
+            currentFileChannel = rf.getChannel();
```

Comment:
Should this temp file channel be closed at some point? Is it closed as a part of random access file closing?

Reply:
This file channel is closed along with `RandomAccessFile` closing.